hack/lib/build/binaries.sh

```diff
@@ -187,16 +190,25 @@ os::build::internal::build_binaries() {
 
     os::build::export_targets "$@"
 
+    if [[ ! "${targets[@]:+${targets[@]}}" || ! "${binaries[@]:+${binaries[@]}}" ]]; then
```

Comment:
`-z` over `!` ?

Reply:
i was having some problem with it here - will try to standardize in a follow up